sklearn/utils/estimator_checks.py

```diff
@@ -3121,7 +3121,8 @@ def check_n_features_in_after_fitting(name, estimator_orig):
     # Make sure that n_features_in are checked after fitting
     tags = _safe_tags(estimator_orig)
 
-    if "2darray" not in tags["X_types"] or tags["no_validation"]:
+    if ("2darray" not in tags["X_types"] and "sparse" not in tags["X_types"] or
+        tags["no_validation"]):
```

Comment:
TfidfTransformer has tags["X_types"] = "sparse". I guess it does not hurt to allow it as well since sparse is 2d